### PR TITLE
(PCP-736) Fix umask

### PR DESCRIPTION
**Pull request by MikaelSmith**

### CODE REVIEW

**Comment by MikaelSmith** [comment]
Testing a variety of platforms at https://jenkins-master-prod-1.delivery.puppetlabs.net/view/puppet-agent/view/ad-hoc/view/vmpooler/job/platform_puppet-agent_init-van-int_suite-ad-hoc-vmpooler-ad-hoc/46/.

**Comment by MikaelSmith** [comment]
Tests failed Solaris, OSX, and Windows for different reasons. I'll continue fixing up tests.

**Comment by MikaelSmith** [comment]
I think I've fixed all the identified issues. I'll rerun on a larger platform set.

**Comment by MikaelSmith** [comment]
Still failing on SLES12, OSX :(

**Comment by MikaelSmith** [comment]
I think the remaining problem is that I haven't figured out how to set the umask on Solaris, SLES, and MacOS. Once those are addressed, this should be ready.

**Comment by MikaelSmith** [comment]
Passes manually, kicking off builds to verify in Jenkins. Starts with https://jenkins-master-prod-1.delivery.puppetlabs.net/view/puppet-agent/view/ad-hoc/view/vmpooler/job/platform_puppet-agent_init-van-int_suite-ad-hoc-vmpooler-ad-hoc/57/.

File: acceptance/tests/pxp-module-puppet/run_puppet_umask.rb [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+  env_name = File.basename(__FILE__, '.*')
+  environment_name = mk_tmp_environment(env_name)
+  test_file = "/tmp/#{environment_name}"
+
+  step 'On master, create a new environment that creates a file' do
+    site_manifest = "#{environmentpath}/#{environment_name}/manifests/site.pp"
+    create_remote_file(master, site_manifest, <<-SITEPP)
+node default {
+  exec {'/bin/touch #{test_file}':}
```
Comment (mruzicka) [inline acceptance/tests/pxp-module-puppet/run_puppet_umask.rb]: This fails on OS X's because `touch` lives in `/usr/bin` there.

**Pull request merged**